processor/error/payload.go

```diff
@@ -18,9 +19,45 @@ type payload struct {
 	Service m.Service
 	System  *m.System
 	Process *m.Process
-	User    common.MapStr
+	User    *m.User
 
-	Events []Event `mapstructure:"errors"`
+	Events []Event
+}
+
+func (pa *payload) decode(raw map[string]interface{}) error {
+	if err := pa.Service.Decode(raw["service"]); err != nil {
```

Comment:
I think all this `if` boilerplate is much simplified if you use functions instead of methods; and propagate error through calls.

eg,  for system:

```
func DecodeSystem(input interface{}, err error) (*System, error) {
	if err != nil {
		return nil, err
	}
	raw, _ := input.(map[string]interface{})
	df := utility.DataFetcher{}
	return &System{
		Hostname: df.StringPtr(raw, "hostname"),
		Platform: df.StringPtr(raw, "platform"),
		Architecture: df.StringPtr(raw, "architecture"),
		IP: df.StringPtr(raw, "ip"),
	}, df.Err
}
```

Then in `decode` you can do just like this:

```
var err error
pa.Service, err = m.DecodeService(raw["system"], err)
pa.System, err = m.DecodeSystem(raw["system"], err)
pa.Process, err = m.DecodeProcess(raw["process"], err)
...
return err
```